### PR TITLE
EORG Arena Prototype [MDB IGNORE]

### PR DESCRIPTION
**Pull request by francinum**
Co-authored-by: Sinestia <Sinestia@users.noreply.github.com>

## About The Pull Request
EORG is now segmented from the main station state

## Why It's Good For The Game
Segments EORG Players from Non-EORG Players. Roleplay thing.

## Changelog
:cl:
tweak: Players can now opt out of "end of round grief" by toggling the, "Go to the post-round arena" preference. The default is on.
/:cl:

### CODE REVIEW

**Comment by Sinestia** [comment]
Partially verified is Good Enough you don't need my GPG key

**Comment by BrumalLake** [comment]
oho now we get to Roleplay post-round
the RP never ends

**Comment by Sinestia** [comment]
> 
> 
> oho now we get to Roleplay post-round
> the RP never ends

Optionally the RP never ends tm

**Comment by Vexylius** [comment]
If I can suggest anything, I would instead of teleporting people to an arena, I would spawn them all with the same kit so that powergaming for EORG is impossible.

...or maybe you lose everything **but** your hat? 👀

File: code/modules/client/preferences_toggles.dm [hunk 1]
```diff
@@ -195,6 +195,13 @@
 	to_chat(usr, "You will [(prefs.chat_toggles & CHAT_BANKCARD) ? "now" : "no longer"] be notified when you get paid.")
 	SSblackbox.record_feedback("nested tally", "preferences_verb", 1, list("Toggle Income Notifications", "[(prefs.chat_toggles & CHAT_BANKCARD) ? "Enabled" : "Disabled"]"))
 
+/client/verb/toggle_eorg_arena()
+	set name = "Toggle Post-Round Arena"
+	set category = "Preferences"
+	set desc = "Toggles your participation in the Post-Round Arena"
+	prefs.eorg_arena = !prefs.eorg_arena
+	to_chat(usr, "You will [(prefs.eorg_arena) ? "now" : "no longer"] take part in the Post-Round Arena.)")
```
Comment (mcmeiler) [inline code/modules/client/preferences_toggles.dm]:
```suggestion
	to_chat(usr, "You will [(prefs.eorg_arena) ? "now" : "no longer"] take part in the Post-Round Arena.")
```
Currently, it writes to chat like this: 
grafik

**Review by mcmeiler — CHANGES_REQUESTED** [review]

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -550,6 +551,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 			dat += "<b>PDA Style:</b> <a href='?_src_=prefs;task=input;preference=pda_style'>[pda_style]</a><br>"
 			dat += "<br>"
 			dat += "<b>Crew Objectives:</b> <a href='?_src_=prefs;preference=crewobj'>[(crew_objectives) ? "Yes" : "No"]</a><br>"
+			dat += "<b>Go to the Post-Round Arena?:</b> <a href='?_src_=prefs;preference=eorg_arena'>[(eorg_arena) ? "Yes" : "No"]</a><br>"
```
Comment (Crossedfall) [inline code/modules/client/preferences.dm]:
```suggestion
			dat += "<b>Go to the post-round arena?:</b> <a href='?_src_=prefs;preference=eorg_arena'>[(eorg_arena) ? "Yes" : "No"]</a><br>"
```

Not sure this is all that clear, but I can't think of anything all that much better. Best I could come up with was:

Opt into end-round grief?

**Comment by francinum** [comment]
Consideration: Slightly increase post-round time.

**Comment by SergeiKoralev** [comment]
It's nice not getting your eyes stabbed out with a screwdriver after shift. Let's send the most bloodthirsty of us away!

**Comment by yyzsong** [comment]
Now I don't get to beat the snot out of that one jackass I don't like after round end! Literally 1984!!!

**Comment by MCterra10** [comment]
why is this needed

**Comment by PowerfulBacon** [comment]
This doesn't really need MDB ignore, theres only 1 map change

**Comment by PowerfulBacon** [comment]
Also horray now I get to do the amazing RP of standing in an empty room for 90 seconds or teleported to a death arena, can't wait to see the improved RP standards 👍 .

**Comment by FriendlyContractor** [comment]
I cannot see why would anybody want this. End of the round grief is a very long standing tradition and im certain overwhelming majority of players would not like for it to be removed

**Comment by ivanmixo** [comment]
> I cannot see why would anybody want this. End of the round grief is a very long standing tradition and im certain overwhelming majority of players would not like for it to be removed

The people who want to participate get an improved experience, the ones who don't also get an improved experience. Win-win imo.

**Comment by AnCopper** [comment]
It works on TGMC.

**Comment by williamcll** [comment]
I don't understand, what happens to the players that have EORG turned off?

**Comment by Sinestia** [comment]
> 
> 
> I don't understand, what happens to the players that have EORG turned off?

nothing

**Comment by Sinestia** [comment]
> 
> 
> I cannot see why would anybody want this. End of the round grief is a very long standing tradition and im certain overwhelming majority of players would not like for it to be removed

They can always turn it on, and EORG has been something a decent amount of people complain about too. 

Especially administratively, a lot of people do it too early due to not actually knowing when the round's "over" (which isn't the moment the shuttle docks, its right after.) Plus there's the issue of EORG prepping. Both of which can get you banned already but in addition to player complaints, it makes sense why this PR exists. 

To be fair though, this is definitely a compromise and only helps solve one of those issues. The repugnant conclusion is that EORG doesn't actually belong on a server seriously focused on RP, but I like it a lot and want it to stay.

**Comment by francinum** [comment]
The three core reasons I did this:

- It was agreed to as the best reasonable solution internally
- It positively defines the limits of EORG instead of requiring prior knowledge
- I'll probably end up wiping people's outfits and selecting from a pool of them to make EORG prep impossible

The map is a meme and should absolutely be replaced. It was thrown together as a cursed joke.

**Comment by BrumalLake** [comment]
why would you want this
NO MORE FUCKING ROBOTICISTS BUILDING DURANDS WITHOUT HOS PERMISSION OR SCIENTISTS MAKING MAXCAPS JUST FOR EORG
well actually i dunno how this works with mechs and whether you can take carry ons to the arena
well at least no shits prematurely detonating

**Comment by francinum** [comment]
> why would you want this
> NO MORE FUCKING ROBOTICISTS BUILDING DURANDS WITHOUT HOS PERMISSION OR SCIENTISTS MAKING MAXCAPS JUST FOR EORG
> well actually i dunno how this works with mechs and whether you can take carry ons to the arena

It rips you out of mechs assuming the code works how I expect it to. Not actually something I've tested.
Right now if you have a mob you're just teleported to the arena, if you're an AI it makes you a syndicate cyborg, otherwise it turns you into an assistant.

**Comment by utilbird** [comment]
you hate to see it

**Review by Sinestia — APPROVED** [review]
Looks alright, post-round arena still feels slightly odd but "EORG Arena" also means nothing to new people.

**Comment by ike709** [comment]
Fran I'll bully you if this goes poorly.

**Comment by KubeRoot** [comment]
Something to consider, if not already implemented, would be to give players that opted out of EORG a button so they can join in on it later, in case roundend stretches a bit and they decide they want to click spessmen horizontal while they wait.

**Comment by francinum** [comment]
wyci

**Comment by Vexylius** [comment]
truly, an end of an era

File: code/modules/client/preferences_savefile.dm [hunk 3]
```diff
@@ -141,6 +141,8 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	if(current_version < 27)
 		if (!(underwear in GLOB.underwear_list))
 			underwear = "Nude"
+	if(current_version < 36)
+		eorg_arena = TRUE //Burn a version to force this on.
```
Comment (PowerfulBacon) [inline code/modules/client/preferences_savefile.dm:145]: This is pointless, save file bumping is for situations where the default value needs to be changed. The default value is already TRUE

File: code/__HELPERS/roundend.dm [hunk 4]
```diff
@@ -269,6 +270,43 @@
...
+					if(/mob/dead/observer)
+						var/mob/living/carbon/human/jonesy = new(dest)
+						jonesy.equipOutfit(/datum/outfit/job/assistant)
+						jonesy.client = C
+						continue
+					if(/mob/dead/new_player)
+						continue //Don't touch new players.
+					//Give AIs a reward for suffering through the round being an AI.
+					if(/mob/living/silicon/ai)
+						var/mob/living/silicon/robot/modules/syndicate/borgie = new(dest)
```
Comment (PowerfulBacon) [inline code/__HELPERS/roundend.dm:298]: This seems a lot more powerful considering nobody in the arena gets flash protection or any weapons at all

File: code/__HELPERS/roundend.dm [hunk 5]
```diff
@@ -269,6 +270,43 @@
 	if(CONFIG_GET(flag/automapvote))
 		SSvote.initiate_vote("map", "BeeBot", forced=TRUE, popup=TRUE) //automatic map voting
 
+	if(eorg_clients)
```
Comment (PowerfulBacon) [inline code/__HELPERS/roundend.dm:273]: this should really be its own proc and not clutter up the round end menu

**Review by PowerfulBacon — COMMENTED** [review]
1. This should be a config option
2. There are a few serious exploits related to it, see #development

File: code/__HELPERS/roundend.dm (hunk 4 above)
Comment (PowerfulBacon) [reply code/__HELPERS/roundend.dm:298]: (balance issue rather than an actual issue)

**Comment by PowerfulBacon** [comment]
Also a bunch of people keep their items

**Comment by PowerfulBacon** [comment]
This is extremely poorly implemented

**Comment by ike709** [comment]
Alright time to bully Fran

File: code/modules/client/preferences_savefile.dm (hunk 3 above)
Comment (francinum) [reply code/modules/client/preferences_savefile.dm:145]: I ran it past crossed.

**Comment by francinum** [comment]
Yeah no nevermind I have no interest in fixing this.

**Pull request merged**